src/test/java/com/onlinebookstore/store/controllertests/CategoryControllerTest.java

```diff
@@ -0,0 +1,149 @@
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Sql(scripts = "classpath:database/test/categories/add-categories-to-table.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@Sql(scripts = "classpath:database/test/books/delete-books-categories.sql",
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@Sql(scripts = "classpath:database/test/categories/delete-categories.sql",
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+class CategoryControllerTest {
+
+    private static MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("Create Category")
+    void createCategory_ValidRequest_ShouldSaveCategory_Ok() throws Exception {
+        CategoryResponseDto createCategoryDto = new CategoryResponseDto()
+                .setName("New Category")
+                .setDescription("Category description");
+
+        String jsonRequest = objectMapper.writeValueAsString(createCategoryDto);
+
+        MvcResult result = mockMvc.perform(
+                        post("/categories")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        CategoryResponseDto actualDto = objectMapper.readValue(result.getResponse()
+                        .getContentAsString(),
+                CategoryResponseDto.class);
+        Assertions.assertEquals(createCategoryDto
+                        .getName(),
+                actualDto.getName());
+        Assertions.assertEquals(createCategoryDto
+                        .getDescription(),
+                actualDto.getDescription());
+    }
+
+    @Test
+    @WithMockUser(username = "user", roles = {"USER"})
+    @DisplayName("Get all categories")
+    void getAllCategories_ShouldReturnCategoriesList_Ok() throws Exception {
+        MvcResult result = mockMvc.perform(get("/categories")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        CategoryResponseDto[] actual = objectMapper.readValue(result.getResponse()
+                        .getContentAsString(),
+                CategoryResponseDto[].class);
+        Assertions.assertTrue(actual.length > 0);
```

Comment:
Will be better to check if we have result with exactly the same size as we expect